src/processFile.ts

```diff
@@ -102,32 +98,34 @@ function adjustImportExportStatements(src: SourceFile, options: ProcessFileOptio
                 }
             }
             const newRef = calcRelativeImportFilename(reference, srcFilename, srcRoot, targetRoot, ext);
-            magicString.update(start, end, newRef);
             linesChanged += 1;
-            continue;
+            return match.replace(reference, newRef);
         }
+        return match;
+    });
+
+    // Apply all changes at once
+    if (linesChanged > 0) {
+        magicString.update(0, content.length, processedContent);
     }
```

Comment:
```suggestion
```

Applying everything at the end breaks the .map file. The whole point of using magicString is to be able to click on some compiled code and have the editor take you to the source.